include/EVT/dev/encoder.hpp

```diff
@@ -0,0 +1,108 @@
+#ifndef _EVT_ENCODER_
+#define _EVT_ENCODER_
+
+#include <EVT/io/GPIO.hpp>
+#include <EVT/utils/log.hpp>
+#include <EVT/utils/time.hpp>
+
+namespace time = EVT::core::time;
+
+namespace EVT::core::DEV {
+
+/**
+ * This class represents a CTS Series 288 Rotary encoder with a 2-bit Binary resolution.
+ * The static wrappers for the interrupt handles must be created in the main method
+ * see samples/encoder/main.cpp for example implementation.
+ * update() must be called within the main loop to update encoder position.
+ */
+class Encoder {
+public:
+    /**
+     * Constructs an Encoder instance.
+     *
+     * @param[in] a a pin of the encoder
+     * @param[in] b b pin of the encoder
+     * @param range range of the encoder positions
+     * @param initialPosition initial position the encoder is in
+     */
+    Encoder(IO::GPIO& a, IO::GPIO& b, uint64_t range, uint64_t initialPosition, bool rollOver);
+
+    /**
+     * Reads and updates the encoder position.
+     */
+    void update();
+
+    /**
+     * Returns the current absolute position
+     *
+     * @return the current position of the encoder, between 0 and range, inclusive
+     */
+    uint64_t getPosition() const;
+
+    /**
+     * Handles the interrupt generated by pin a. If two interrupts occur too soon after one another the second is ignored.
+     *
+     * @param pin pin that cause the interrupt
+     */
+    void aInterruptHandler(IO::GPIO* pin);
+
+    /**
+     * Handles the interrupt generated by pin b. If two interrupts occur too soon after one another the second is ignored.
+     *
+     * @param pin pin that cause the interrupt
+     */
+    void bInterruptHandler(IO::GPIO* pin);
+
+private:
+    /** GPIO pin a */
+    IO::GPIO& a;
+    /** GPIO pin b */
+    IO::GPIO& b;
+
+    /** position can be between = [0, range] */
+    uint64_t range;
+    /** the current absolute position of the encoder */
+    uint64_t position;
+    /** if the position rolls over when it reaches range or 0 */
+    bool rollOver;
+    /** counter for how many updates have occurred since the encoder was moved */
+    int8_t currentRelPos;
+
+    ///INTERRUPT RELATED VARIABLES
+    /** the change since the last update was called, updated in the interrupt handler */
+    int64_t interruptChange;
+    /** the most recent time in milliseconds since the last interrupt on pin A */
+    uint32_t recentAInterruptTime;
```

Comment:
This is a nitpick, but would this make more sense?
```suggestion
    uint32_t lastAInterruptTime;
```

Reply:
renamed them!